flipperdevices/flipperzero-firmware
# Prevent idle priority threads from potentially starving the FreeRTOS idle task

# What's new

This PR consists of three commits:
* `FuriThreadPriorityIdle` has been bumped down from `1` to `0`, to be on the same priority as the FreeRTOS idle task. This allows idle priority threads to spinloop (with `furi_thread_yield()`) and not starve that idle task, which apps like the Direct Draw debug app or [Flipper95](https://lab.flipper.net/apps/flipper95) were previously doing. This would manifest itself by qFlipper causing future USB connections to lock up until the offending application has been closed. With this change, the idle task is given enough time to process task deletions, and prevents the lockup.
* `furi_thread_list_process` was made private, because it requires a `runtime` parameter that can only be obtained from FreeRTOS.
* Fixed a memory leak in the DirectDraw debug app, and made it not include `canvas_i.h` that wasn't needed, and made building with uFBT impossible until this include was removed.

API version bump is needed because of `furi_thread_list_process`, but also because of the change to the `FuriThreadPriority` enum.

Fixes #3908

# Verification 

1. Run Direct Draw app or [Flipper95](https://lab.flipper.net/apps/flipper95).
2. Run qFlipper, observe it shows the screen output, then close it.
3. Attempt to run `./fbt cli`. Observe that:
   * Without this PR - the command gets stuck at
      ```
      --- Miniterm on COM6  230400,8,N,1 ---
      --- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
      ```
      until the app is closed.
   * With this PR - the command finishes successfully, showing the full prompt.

# Checklist (For Reviewer)
